client/onion_client.py

```diff
@@ -0,0 +1,144 @@
+import argparse             # for command-line argument parsing
+def shut_down_directory_node():
+    LISTEN_SOCKET.close()
+    if args.verbose: print('Client Node DOWN')
+
+import random               # for random path selection
+import socket               # for TCP communication
+import json                 # for encoding data sent through TCP
+import threading            # for one thread per TCP connection
+
+
+#   CLI ARGS
+###############################################################################
+
+parser = argparse.ArgumentParser()
+
+parser.add_argument("destination", help="hostname to connect to")
+parser.add_argument("port", help="port to listen on", type=int)
+parser.add_argument("-v", "--verbose",
+                    help="level of logging verbose", action="store_true")
+
+args = parser.parse_args()
+
+# validate args against conditions
+if args.port < 5551 or args.port > 5557: # 7 group members, each get a port
+    parser.error("port must satisfy: 5551 <= port <= 5557")
+
+
+#   GLOBALS
+#################################################################################
+
+HOST = ''                            # empty string => all IPs of this machine
+PORT = args.port                     # port for server to listen on, cli arg
+
+DIRECTORY_NODE = 'cs-1'              # TODO hardcoded value or CL argument
+
+LISTEN_SOCKET = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+LISTEN_SOCKET.bind((HOST, PORT))
+LISTEN_SOCKET.listen(1)
```

Comment:
the `LISTEN_SOCKET` is not used, we don't need it

Reply:
Fixed in  2dadc05